servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -374,7 +373,7 @@ private boolean setFailure0(Throwable cause) {
             // ignore any further results. For non-reliable protocols, when we support them, we will modify this
             // behavior as appropriate.
             if (hasFlag(SUBSCRIBER_TERMINATED)) {
-                return false;
+                return nettySharedPromiseTryStatus();
```

Comment:
this would occasionally contribute to additional overhead and test failures noticed when running on windows (e.g. write failure, netty ChannelOutboundBuffer attempts to fail all queued promise, but may already be failed, then log stack trace). I'm not sure there is a "great" answer to comply with Netty's APIs using the shared promise approach (e.g. using ref counting here is a way to maybe improve, but doesn't solve all issues). open to alternative suggestions.

Reply:
moved to https://github.com/apple/servicetalk/pull/978